src/axom/inlet/examples/user_defined_type.cpp

```diff
@@ -76,21 +78,48 @@ struct FromInlet<LinearSolver>
   }
 };
 
+struct BoundaryCondition
+{
+  std::unordered_map<int, int> attrs;
+  double constant;
+  static void defineSchema(inlet::Table& schema)
+  {
+    schema.addIntArray("attrs", "List of boundary attributes");
+    schema.addDouble("constant",
+                     "The scalar to fix the value of the solution to");
+  }
+};
+
+template <>
+struct FromInlet<BoundaryCondition>
+{
+  BoundaryCondition operator()(inlet::Table& base)
+  {
+    BoundaryCondition bc;
+    bc.attrs = base["attrs"];
+    bc.constant = base["constant"];
+    return bc;
+  }
+};
+
 struct ThermalSolver
 {
   Mesh mesh;
   LinearSolver solver;
+  std::unordered_map<int, BoundaryCondition> bcs;
   // defineSchema is intended to be used recursively
   // Tables are created for subobjects and passed to
   // subobject defineSchema implementations
-  static void defineSchema(SchemaCreator& schema)
+  static void defineSchema(inlet::Table& schema)
   {
     auto mesh_table = schema.addTable("mesh", "Information about the mesh");
     Mesh::defineSchema(*mesh_table);
     auto solver_table =
       schema.addTable("solver",
                       "Information about the iterative solver used for Ku = f");
     LinearSolver::defineSchema(*solver_table);
+    auto bc_table = schema.addGenericArray("bcs", "List of boundary conditions");
+    BoundaryCondition::defineSchema(*bc_table);
```

Comment:
I am concerned a user would not understand how this works... more comments are needed like "addGenericArray loops over found indices and creates array of boundary conditions"

Maybe there should a snippet of an example lua in each defineSchema?